src/client/cli/formatter/table_formatter.cpp

```diff
@@ -53,91 +53,150 @@ std::string to_usage(const std::string& usage, const std::string& total)
     return fmt::format("{} out of {}", mp::MemorySize{usage}.human_readable(), mp::MemorySize{total}.human_readable());
 }
 
-std::string generate_instance_info_report(const mp::InfoReply& reply)
+std::string generate_snapshot_details(const mp::DetailedInfoItem& item)
 {
     fmt::memory_buffer buf;
+    const auto& fundamentals = item.snapshot_info().fundamentals();
+
+    fmt::format_to(std::back_inserter(buf), "{:<16}{}\n", "Snapshot:", fundamentals.snapshot_name());
+    fmt::format_to(std::back_inserter(buf), "{:<16}{}\n", "Instance:", item.name());
+    fmt::format_to(std::back_inserter(buf), "{:<16}{}\n",
+                   "Size:", item.snapshot_info().size().empty() ? "--" : item.snapshot_info().size());
+    fmt::format_to(std::back_inserter(buf), "{:<16}{}\n", "CPU(s):", item.cpu_count());
+    fmt::format_to(std::back_inserter(buf), "{:<16}{}\n", "Disk space:", item.disk_total());
+    fmt::format_to(std::back_inserter(buf), "{:<16}{}\n", "Memory size:", item.memory_total());
+
+    auto mount_paths = item.mount_info().mount_paths();
+    fmt::format_to(std::back_inserter(buf), "{:<16}{}", "Mounts:", mount_paths.empty() ? "--\n" : "");
+    for (auto mount = mount_paths.cbegin(); mount != mount_paths.cend(); ++mount)
+    {
+        if (mount != mount_paths.cbegin())
+            fmt::format_to(std::back_inserter(buf), "{:<16}", "");
+        fmt::format_to(std::back_inserter(buf), "{:{}} => {}\n", mount->source_path(),
+                       item.mount_info().longest_path_len(), mount->target_path());
+    }
 
-    for (const auto& info : mp::format::sorted(reply.detailed_report().details()))
+    fmt::format_to(std::back_inserter(buf), "{:<16}{}\n",
+                   "Created:", google::protobuf::util::TimeUtil::ToString(fundamentals.creation_timestamp()));
+    fmt::format_to(std::back_inserter(buf), "{:<16}{}\n",
+                   "Parent:", fundamentals.parent().empty() ? "--" : fundamentals.parent());
+
+    auto children = item.snapshot_info().children();
+    fmt::format_to(std::back_inserter(buf), "{:<16}{}", "Children:", children.empty() ? "--\n" : "");
+    for (auto child = children.cbegin(); child != children.cend(); ++child)
     {
-        const auto& instance_details = info.instance_info();
+        if (child != children.cbegin())
+            fmt::format_to(std::back_inserter(buf), "{:<16}", "");
+        fmt::format_to(std::back_inserter(buf), "{}\n", *child);
+    }
 
-        fmt::format_to(std::back_inserter(buf), "{:<16}{}\n", "Name:", info.name());
-        fmt::format_to(std::back_inserter(buf), "{:<16}{}\n",
-                       "State:", mp::format::status_string_for(info.instance_status()));
-        fmt::format_to(std::back_inserter(buf), "{:<16}{}\n", "Snapshots:", instance_details.num_snapshots());
+    fmt::format_to(std::back_inserter(buf), "{:<16}{}\n",
+                   "Comment:", fundamentals.comment().empty() ? "--" : fundamentals.comment());
```

Comment:
Can we pad extra lines with 16 spaces if the comment is multi-line, please?

Reply:
Does `fmt` support this? I'm unable to find any solution for this other than manually splitting the string based on terminal width and aligning each line separately.

Reply:
Hmm, I see what you mean. I don't think `fmt` can do that. What I had in mind was just to prepend spaces to existing newlines in the string. That can be achieved with substring replacement. To account for Windows, we need to replace `\r\n` and `\n` when not preceded by `\r`.

Longer lines will still wrap around, but perhaps that is OK for now? WDYT?